jep/0000/README.adoc

```diff
@@ -0,0 +1,358 @@
+= JEP-0000: End Java 8 support in Jenkins
```

Comment:
I do not think the phrasing in the title is clear enough. It makes it sound like we will close bug reports related to Java 8 as “will not fix” but you could still kind of keep things going if you did not want “support”. That is not what is being proposed here IIUC, or at least not what the key https://github.com/jenkinsci/jenkins/pull/6083 does: it means that subsequent Jenkins releases will not run on Java 8, at all. `/opt/jdk8/bin/java -war jenkins.war` will just print an error about unrecognized bytecode version and immediately exit. So it would be clearer to emphasize that the change is to require Java 11 (or newer I suppose) to run.

(IIRC previous Java upgrades involved some patches to `executable-war` and/or `remoting` to compile the entry point class with an older bytecode version so that it could detect the JRE version and print a nicer error.)

Also BTW the PR title (but not the JEP title) talks about Jenkins _core_, which is accurate only to a certain extent—plugins may continue to “support” Java 8 indefinitely, but only insofar as they avoid updating their core baseline (and perhaps also avoid updating the parent POM, TBD).

Reply:
I like the insight that the title is not clear enough.  Here are some alternatives for consideration

State the new minimum Java version instead of stating the end of support of Java 8

= JEP-0000: Require Java 11 as minimum Java version

Alternatively, a shorter form of the same idea

= JEP-0000: Require Java 11 or newer

Another alternative, state the technical change

= JEP-0000: Use Java 11 bytecode instead of Java 8 bytecode

I'm open to other alternatives as well.

Reply:
> Use Java 11 bytecode instead of Java 8 bytecode

Misleading, because we are also requiring Java 11 API surface, not merely bytecode.

> Require Java 11 or newer

I like this. Succinct and emphatic.

Reply:
Fixed title in 2acc2f6f0a7d1c5889a98ce13d272c15495a38e6

Reply:
Fixed in 718a3ac79b2b2ffc7ad68e59dff5c1f9a7fa09fe